nuxeo-runtime/nuxeo-connect-standalone/src/main/java/org/nuxeo/launcher/connect/ConnectBroker.java

```diff
@@ -1253,6 +1252,10 @@ public boolean pkgRequest(List<String> pkgsToAdd, List<String> pkgsToInstall, Li
                         namesOrIdsToInstall, localSnapshotsToUninstall, localSnapshotsToReplace,
                         localSnapshotsToMaybeReplace);
 
+                // Exclude snapshot packages which are not available in remote
+                localSnapshotsToMaybeReplace.removeIf(pkg -> !isRemotePackageId(pkg));
+                localSnapshotsToReplace.removeIf(pkg -> !isRemotePackageId(pkg));
```

Comment:
I agree with you. However, I cannot find out a test case to prove that a local-only snapshot package must delete from `localSnapshotsToUninstall ` . Could you help me to define one?